packages/shared/src/components/widgets/PostUsersHighlights.tsx

```diff
@@ -116,30 +135,47 @@ const UserHighlight = (props: SourceAuthorProps) => {
         />
       )}
       <ConditionalWrapper
-        condition={userTypeNotSource}
+        condition={!isUserTypeSource}
         wrapper={(children) => (
           <ProfileTooltip user={{ id }}>
             {children as ReactElement}
           </ProfileTooltip>
         )}
       >
-        <div className="ml-4 flex flex-col">
-          <div className="flex">
-            <ProfileLink className="font-bold typo-callout" href={permalink}>
+        <div className="ml-4 flex min-w-0 flex-1 flex-col">
+          {isUserTypeSource && (
+            <ProfileLink
+              className={classNames('!block truncate font-bold typo-callout')}
+              href={permalink}
+            >
               {name}
             </ProfileLink>
-            <ReputationUserBadge user={{ reputation }} />
-          </div>
+          )}
+          {!isUserTypeSource && (
+            <div className="flex">
+              <ProfileLink className="font-bold typo-callout" href={permalink}>
+                {name}
+              </ProfileLink>
+              <ReputationUserBadge user={{ reputation }} />
+            </div>
+          )}
```

Comment:
I think a ternary would be better here, since we are basically reversing the condition:
```jsx
          {isUserTypeSource ? (
            <ProfileLink className="font-bold typo-callout" href={permalink}>	            <ProfileLink
              className={classNames('!block truncate font-bold typo-callout')}
              href={permalink}
            >
              {name}	
            </ProfileLink>
          ) : (
            <div className="flex">
              <ProfileLink className="font-bold typo-callout" href={permalink}>
                {name}
              </ProfileLink>
              <ReputationUserBadge user={{ reputation }} />
            </div>
          )}
```

Reply:
I don't really like ternaries in JSX. Hard to follow and nest. It is also experiment so it will be cleaned up a bit when either one wins.

Reply:
If it weren't for the extra classNames, we could've used ConditionalWrapper
```tsx
<ConditionalWrapper
  condition={!isUserTypeSource}
  wrapper={(component) => (
    <div className="flex">
      {component}
      <ReputationUserBadge user={{ reputation }} />
    </div>
  )}
>
```

Maybe it might work if `component` is `Component` with the props added? Don't know if that'll work.
Could also just add that onto the ProfileLink itself

```tsx
<ConditionalWrapper
  condition={!isUserTypeSource}
  wrapper={(component) => (
    <div className="flex">
      {component}
      <ReputationUserBadge user={{ reputation }} />
    </div>
  )}
>
  <ProfileLink
    className={classNames(
      "font-bold typo-callout",
      isUserTypeSource && "!block truncate")}
    href={permalink}
  >
    {name}
  </ProfileLink>
</ConditionalWrapper>
```

Reply:
Did not manage truncate to work with reputation badge. I don't have the nerve to play with CSS that much.

Reply:
truncating is... yeah... I've done a few PRs this week with them 😿